src/services/superfluidMinionService.js

```diff
@@ -0,0 +1,142 @@
+import Web3 from 'web3';
+
+import SuperfluidMinionAbi from '../contracts/superfluidMinion.json';
+import { SUPERFLUID_MINION_STREAMS } from '../graphQL/superfluid-queries';
+import { TokenService } from './tokenService';
+import { chainByID, getGraphEndpoint } from '../utils/chain';
+import { graphFetchAll } from '../utils/theGraph';
+
+import SuperfluidSDK from '@superfluid-finance/js-sdk';
+
+let sf = null;
+
+const getSuperTokenBalances = async (chainID, minion, superTokens) => {
+  const tokenBalances = superTokens.map(async (tokenAddress) => {
+    const tokenService = TokenService({
+      tokenAddress: tokenAddress,
+      chainID,
+    });
+    return {
+      [tokenAddress]: {
+        tokenBalance: await tokenService('balanceOf')(minion),
+        symbol: await tokenService('symbol')(minion),
+        decimals: await tokenService('decimals')(minion),
+      },
+    };
+  });
+  return Object.assign({}, ...(await Promise.all(tokenBalances)));
+};
+
+export const SuperfluidMinionService = ({ web3, minion, chainID }) => {
+  const chainConfig = chainByID(chainID);
+  if (!web3) {
+    const rpcUrl = chainConfig.rpc_url;
+    web3 = new Web3(new Web3.providers.HttpProvider(rpcUrl));
+  }
+
+  const contract = new web3.eth.Contract(SuperfluidMinionAbi, minion);
+
+  return function getService(service) {
+    if (service === 'fetchStreams') {
+      return async ({ molochAddress }) => {
+        const streams = await graphFetchAll({
+          endpoint: getGraphEndpoint(chainID, 'subgraph_url'),
+          query: SUPERFLUID_MINION_STREAMS,
+          variables: {
+            minionId: `${molochAddress}-minion-${minion}`,
+          },
+          subfield: 'minionStreams',
+        });
+        if (!streams.length) {
+          return {
+            flows: [],
+            superTokens: null,
+          };
+        }
+        const superTokens = await getSuperTokenBalances(
```

Comment:
I'm beginning to notice that this whole pattern doesn't follow the TX pattern in the app. Will need to know why.

It looks like we're returning vals to React to update the UI. This might work if we're not relying on the graph. Will need to deep dive on this one.